dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -416,17 +430,73 @@ def load_file_data(self, sess, bucket_name, region_name):
 
         return file_row_count
 
-    def process_data_chunk(self, sess, chunk_df):
+    def parallel_data_loading(self, reader_obj):
+        with Manager() as server_manager:
+            # These variables will need to be shared among the processes and used later overall
+            shared_data = server_manager.dict()
+            shared_data['total_rows'] = self.total_rows
+            shared_data['has_data'] = self.has_data
+            shared_data['error_rows'] = self.error_rows
+            shared_data['error_list'] = self.error_list
+            shared_data['header_dict'] = self.reader.header_dict
+            shared_data['flex_fields'] = self.reader.flex_fields
+            # setting reader to none as multiprocess can't pickle it, it'll get reset
+            temp_reader = self.reader
+            self.reader = None
+
+            m_lock = server_manager.Lock()
+            pool = Pool(MULTIPROCESSING_POOLS)
+            for chunk_df in reader_obj:
+                pool.apply_async(func=self.process_data_chunk, args=(chunk_df, shared_data, m_lock))
+            pool.close()
+            pool.join()
+
+            # Resetting these out here as they are used later in the process
+            self.total_rows = shared_data['total_rows']
+            self.has_data = shared_data['has_data']
+            self.error_rows = shared_data['error_rows']
+            self.error_list = shared_data['error_list']
+            self.reader = temp_reader
+
+    def iterative_data_loading(self, reader_obj):
+        shared_data = {
```

Comment:
stylistically, would be better to see this dictionary and the same one in the parallel function being set the same way (curly assignment, or ['indexed'] = val assignment.

Reply:
They're technically two different objects but I can restyle them so they look initialized in the same way.